feature_engine/selection/drop_psi_features.py

```diff
@@ -0,0 +1,537 @@
+from typing import List, Union
+
+import numpy as np
+import pandas as pd
+
+from feature_engine.dataframe_checks import (
+    _check_contains_inf,
+    _check_contains_na,
+    _is_dataframe,
+)
+from feature_engine.discretisation import (
+    EqualFrequencyDiscretiser,
+    EqualWidthDiscretiser,
+)
+from feature_engine.selection.base_selector import BaseSelector
+from feature_engine.variable_manipulation import (
+    _check_input_parameter_variables,
+    _find_or_check_numerical_variables,
+)
+
+Variables = Union[None, int, str, List[Union[str, int]]]
+
+
+class DropHighPSIFeatures(BaseSelector):
+    """
+    DropHighPSIFeatures drops features with a Population Stability Index (PSI) above
+    a given threshold. The PSI of a feature is an indication of the shift in its
+    distribution; a feature with high PSI might therefore be seen as instable.
+
+    In fields like Credit Risk Modelling, the elimination of features with high PSI
+    is frequent and usually required by the Regulator.
+
+    The calculation of the PSI requires to compare two distributions.
+    In DropHighPSIFeatures the approach implemented is the following:
+
+    - The dataframe is split in two parts and the PSI calculations are performed by
+    comparing the distribution of the features in the two parts.
+
+    - The user defines the columns of the dataframe according to which the dataframe
+    will de split. The column can contain numbers, dates and strings.
```

Comment:
I understand the concept of dividing based on dates: we want to determine if a feature changes its distribution in time. This is what is normally asked in finance.

I understand the concept of dividing based on a numerical column, only if the numerical column is the index, and assuming that the index is ordered somehow and not random. When else would we use a numerical column to divide the data?

When would we use a numerical column to divide the data? I understand that it is technically possible. But would anyone divide based on categories in a real financial setting?

Reply:
You did, thank you for taking the time to explain :)